src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -184,6 +184,9 @@ internal bool IsScriptFileDetermined(FunctionMetadata functionMetadata)
 
         private void LoadCustomProviderFunctions(List<FunctionMetadata> functionMetadataList)
         {
+            // We always want to get the most updated function providers in case this list was changed.
+            _functionProviders = _serviceProvider.GetService<IEnumerable<IFunctionProvider>>();
```

Comment:
Doesn't look like there's really a need for this to be a class variable. Can you make this a local and modify the `AddMetadataFromCustomProviders` method to take it as an argument as opposed to relying on the variable? That will make this cleaner and ensure you're not working with a cached instance (avoiding accidental use of the class variable)

Reply:
Yeah that sounds like a good idea. Thanks! Will update.